From 28c30d597abc3c02eb01d2d984c1084f45a7fccd Mon Sep 17 00:00:00 2001
From: Cheng Yao <yao.cheng@intel.com>
Date: Fri, 29 Mar 2013 12:38:01 +0800
Subject: VideoEdit: cancel ProgressCb in pause_l to avoid possible dead locking

BZ: 93006

during stopPreview, there's a possible dead lock between
PreviewPlayer::onProgressCbEvent and PreviewPlayer::reset.
This change cancels progressCbEvent in pause_l to avoid
dead lock.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: I91fb1509b5851d749266538ea0b40c37ec8543de
Orig-Change-Id: Ia2525dece7bc96d4b22319152001f9253247346a
Signed-off-by: Cheng Yao <yao.cheng@intel.com>
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/lvpp/PreviewPlayer.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libvideoeditor/lvpp/PreviewPlayer.cpp b/libvideoeditor/lvpp/PreviewPlayer.cpp
index 4d41615..f547260 100755
--- a/libvideoeditor/lvpp/PreviewPlayer.cpp
+++ b/libvideoeditor/lvpp/PreviewPlayer.cpp
@@ -1743,7 +1743,7 @@ status_t PreviewPlayer::pause_l(bool at_eos) {
         return OK;
     }
 
-    cancelPlayerEvents_l();
+    cancelPlayerEvents_l(true);
 
     if (mAudioPlayer != NULL && (mFlags & AUDIO_RUNNING)) {
         if (at_eos) {
-- 
1.7.4.1

